phoenix-core/src/main/java/org/apache/phoenix/iterate/DeferredByteBufferSegmentQueue.java

```diff
@@ -0,0 +1,123 @@
+package org.apache.phoenix.iterate;
+
+import org.apache.commons.io.output.DeferredFileOutputStream;
+import org.apache.phoenix.memory.MemoryManager;
+import org.apache.phoenix.memory.MemoryManager.MemoryChunk;
+
+import java.io.*;
+import java.util.*;
+
+public abstract class DeferredByteBufferSegmentQueue<T> extends BufferSegmentQueue<T> {
+
+    final MemoryChunk chunk;
+
+    public DeferredByteBufferSegmentQueue(int index, int thresholdBytes,
+                                          boolean hasMaxQueueSize, MemoryManager memoryManager) {
+        super(index, thresholdBytes, hasMaxQueueSize);
+        chunk = memoryManager.allocate(thresholdBytes);
```

Comment:
FYI, SpoolingResultIterator is deprecated, so no need to refactor. Feel free to copy/paste code from there, though, as conceptually it's what you need.